File: go/space-age/space_age.go
package space

// Planet defines a planet name
type Planet string

const earthYear = 365.25 * 24 * 60 * 60

var yearOn = map[Planet]float64{
	"Earth":   1,
	"Mercury": 0.2408467,
	"Venus":   0.61519726,
	"Mars":    1.8808158,
	"Jupiter": 11.862615,
	"Saturn":  29.447498,
	"Uranus":  84.016846,
	"Neptune": 164.79132,
}

// Age calculates how old someone is on a given planet.
func Age(seconds float64, planet Planet) float64 {
	return seconds / (earthYear * yearOn[planet])
}

File: c/space-age/space_age.c
#include "space_age.h"

float age(planet_t planet,
          int64_t seconds) {
  switch (planet) {
  case EARTH:   return seconds / (365.25 * 24 * 60 * 60);
  case MERCURY: return age(EARTH, seconds) / 0.2408467;
  case VENUS:   return age(EARTH, seconds) / 0.6151973;
  case MARS:    return age(EARTH, seconds) / 1.8808158;
  case JUPITER: return age(EARTH, seconds) / 11.862615;
  case SATURN:  return age(EARTH, seconds) / 29.447498;
  case URANUS:  return age(EARTH, seconds) / 84.016846;
  case NEPTUNE: return age(EARTH, seconds) / 164.79132;
  }

  return -1;
}
